pkg/notifiers/slack/handle_request.go

```diff
@@ -99,23 +99,139 @@ func (n *Notifier) HandleRequestEvent(ctx context.Context, log *zap.SugaredLogge
 						RequestorEmail:   userQuery.Result.Email,
 						ReviewURLs:       reviewURL,
 					})
-					err = SendMessageBlocks(ctx, slackClient, approver.Result.Email, msg, summary)
+
+					ts, err := SendMessageBlocks(ctx, slackClient, approver.Result.Email, msg, summary)
 					if err != nil {
 						log.Errorw("failed to send request approval message", "user", usr, zap.Error(err))
 					}
-				}()
+
+					updatedUsr := usr
+					updatedUsr.SlackMessageID = ts
+					log.Infow("updating reviewer with slack msg id", "updatedUsr.SlackMessageID", ts)
+
+					err = n.DB.Put(ctx, &updatedUsr)
+
+					if err != nil {
+						log.Errorw("failed to update reviewer", "user", usr, zap.Error(err))
+					}
+				}(usr)
 			}
+
 			wg.Wait()
 		} else {
 			//Review not required
 			msg := fmt.Sprintf(":white_check_mark: Your request to access *%s* has been automatically approved. Hang tight - we're provisioning the role now and will let you know when it's ready.", ruleQuery.Result.Name)
 			fallback := fmt.Sprintf("Your request to access %s has been automatically approved.", ruleQuery.Result.Name)
-			n.SendDMWithLogOnError(ctx, slackClient, log, userQuery.Result.Email, msg, fallback)
+			_ = n.SendDMWithLogOnError(ctx, slackClient, log, req.RequestedBy, msg, fallback)
 		}
 	case gevent.RequestApprovedType:
 		msg := fmt.Sprintf("Your request to access *%s* has been approved. Hang tight - we're provisioning the access now and will let you know when it's ready.", ruleQuery.Result.Name)
 		fallback := fmt.Sprintf("Your request to access %s has been approved.", ruleQuery.Result.Name)
-		n.SendDMWithLogOnError(ctx, slackClient, log, userQuery.Result.Email, msg, fallback)
+		_ = n.SendDMWithLogOnError(ctx, slackClient, log, req.RequestedBy, msg, fallback)
+
+		// Loop over the request reviewers
```

Comment:
Seeing this code block repeated 3 times, makes me think it should be split out into a function?